File: Assignment 7/1. Employee Class/Employee.java
public class Employee {
    private String firstName;
    private String lastname;
    private double monthySalary;

    public Employee(String firstName, String lastname, double monthySalary) {
        this.firstName = firstName;
        this.lastname = lastname;
        if (monthySalary >= 0) {
            this.monthySalary = monthySalary;
        } else {
            this.monthySalary = 0.0;
        }
    }

    public String getFirstName() {
        return firstName;
    }
    public String getLastName() {
        return lastname;
    }
    public double getMonthySalary() {
        return monthySalary;
    }
    public void setFirstName(String firstName) {
        this.firstName = firstName;
    }
    public void setLastname(String lastname) {
        this.lastname = lastname;
    }
    public void setMonthySalary(double monthySalary) {
        if (monthySalary >= 0) {
            this.monthySalary = monthySalary;
        }
    }
    public double getYearlySalary() {
        // yearly bonus
        return monthySalary * 13;
    }
    public void giveRaise(double raisePercentage) {
        this.monthySalary += monthySalary * raisePercentage/100;
    }
}

File: Employee.cpp
#include <iostream>;
#include <string>;
#include "Employee.h";
int main() {
    Employee emp1("Quang", "Huy", 3000);
    Employee emp2("Obito", "Uchihahaha", 4000);

    cout << "Initial Yearly Salaries:\n";
    cout << emp1.getFirstName() << " " << emp1.getLastName() << ": $" << emp1.getYearlySalary() << endl;
    cout << emp2.getFirstName() << " " << emp2.getLastName() << ": $" << emp2.getYearlySalary() << endl;

    emp1.giveRaise(10);
    emp2.giveRaise(10);

    cout << "\nYearly Salaries After 10% Raise:\n";
    cout << emp1.getFirstName() << " " << emp1.getLastName() << ": $" << emp1.getYearlySalary() << endl;
    cout << emp2.getFirstName() << " " << emp2.getLastName() << ": $" << emp2.getYearlySalary() << endl;

    return 0;
}